### PR TITLE
doc/dev: improve basic-workflow.rst

### PR DESCRIPTION
**Pull request by zdover23**
Make various improvements to doc/dev/developer_guide/basic-workflow.rst.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: doc/dev/developer_guide/basic-workflow.rst [hunk 1]
```diff
@@ -180,12 +180,12 @@ at the moment that you cloned it, but the upstream repo
 that it was forked from is not frozen in time: the upstream repo is still being
 updated by other contributors. 
 
-Because upstream main is continually receiving updates from other
-contributors, your fork will drift farther and farther from the state of the
-upstream repo when you cloned it.
+Because upstream main is continually receiving updates from other contributors,
+over time your fork will drift farther and farther from the state of the
+upstream repo as it was when you cloned it.
```
Comment (anthonyeleven) [inline doc/dev/developer_guide/basic-workflow.rst:185]: repository

File: doc/dev/developer_guide/basic-workflow.rst [hunk 2]
```diff
@@ -243,15 +243,16 @@ Fixing the bug in the local working copy
...
-   This guide assumes that you know what required improvement, and that you
-   know what to do to provide that improvement.
+   This guide assumes that you have identified an area that required
+   improvement, and that you know how to make that improvement.
 
-   It might be that your fix is simple and requires only minimal testing. But
-   that's unlikely. It is more likely that the process of fixing your bug will
-   be iterative and will involve trial, error, skill, and patience. 
+   It might be that your fix is simple and that it requires only minimal
+   testing. But that's unlikely. It is more likely that the process of fixing
```
Comment (anthonyeleven) [inline doc/dev/developer_guide/basic-workflow.rst:250]: unlikely (unless only documentation is updated).

**Review by anthonyeleven — APPROVED** [review]

File: doc/dev/developer_guide/basic-workflow.rst (hunk 1 above)
Comment (zdover23) [reply doc/dev/developer_guide/basic-workflow.rst:185]: Good catch.

File: doc/dev/developer_guide/basic-workflow.rst (hunk 2 above)
Comment (zdover23) [reply doc/dev/developer_guide/basic-workflow.rst:250]: Fair enough.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/58937 - Squid backport
https://github.com/ceph/ceph/pull/58938 - Reef backport
https://github.com/ceph/ceph/pull/58939 - Quincy backport

**Pull request merged**